src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverterBase.cs

```diff
@@ -408,10 +421,13 @@ protected void InsertRegularField<TBuilder>(IndexField field, object value, Json
                 builder.WriteSpatial(fieldId, path,  (CoraxSpatialPointEntry)value);
                 break;
             case ValueType.CoraxDynamicItem:
-                var cdi = (CoraxDynamicItem)value;
-                //we want to unpack item here.
                 var old = builder.ResetList(); // For lists of CreatedField(), we ignoring the list
-                InsertRegularField(cdi!.Field, cdi.Value, indexContext, builder, sourceDocument, out shouldSkip);
+                if (value is CoraxDynamicItem standardCdi)
```

Comment:
We didn't have this check before. Why do we have it now? When this cannot be true?

Reply:
Ping

Reply:
Restored the previous code, this was changed only for debugging purposes.